src/librustc_typeck/diagnostics.rs

```diff
@@ -2387,39 +2387,119 @@ impl Copy for &'static Bar { } // error
 "##,
 
 E0207: r##"
-You declared an unused type parameter when implementing a trait on an object.
-Erroneous code example:
+Type parameters and lifetime parameters of `impl` blocks must
```

Comment:
I don't like the way it is formatted. I propose:

"Type parameters and lifetime parameters of `impl` blocks must either:

- appear in the self type of the impl
- for trait impls, appear in the trait reference
- be bound as an associated type"
"